src/main/java/hudson/util/VersionNumber.java

```diff
@@ -513,16 +513,26 @@ public boolean isNewerThan(VersionNumber rhs) {
         return compareTo(rhs) > 0;
     }
 
+    /**
+     * Returns a digit (numeric component) by its position. Once a non-numeric component is found all remaining components
+     * are also considered non-numeric by this method.
+     *
+     * @param idx Digit position we want to retrieve starting by 1 (not zero).
```

Comment:
@stephenc This was my first intention but the `digit()` behavior is totally wrong. I considered it like a bug and this PR its fix. I never saw a code using this method.